src/Core/Services/Implementations/UserService.cs

```diff
@@ -1369,6 +1369,71 @@ public async Task<bool> VerifySecretAsync(User user, string secret, bool isSetti
         return isVerified;
     }
 
+    public async Task<IEnumerable<(Guid userId, bool twoFactorIsEnabled)>> TwoFactorIsEnabledAsync(IEnumerable<Guid> userIds)
+    {
+        var userDetails = await _userRepository.GetManyDetailsAsync(userIds.ToList());
+        var result = new List<(Guid userId, bool hasTwoFactor)>();
+
+        foreach (var userDetail in userDetails)
+        {
+            var providers = userDetail.GetTwoFactorProviders();
+            var hasTwoFactor = false;
+
+            if (providers != null)
+            {
+                foreach (var provider in providers)
+                {
+                    if (provider.Value?.Enabled ?? false)
+                    {
+                        if (!TwoFactorProvider.RequiresPremium(provider.Key))
+                        {
+                            hasTwoFactor = true;
+                            break;
+                        }
+                        if (userDetail.HasPremiumAccess)
+                        {
+                            hasTwoFactor = true;
+                            break;
+                        }
+                    }
+                }
+            }
```

Comment:
One other suggestion, just to be a little more functional:
```suggestion

            // Get all enabled providers
            var enabledProviderKeys = from provider in providers
                where provider.Value?.Enabled ?? false
                select provider.Key;

            // Find the first provider that is enabled and passes the premium check
            var hasTwoFactor = enabledProviderKeys
                .Select(type => userDetail.HasPremiumAccess || !TwoFactorProvider.RequiresPremium(type))
                .FirstOrDefault(twoFactorEnabled => twoFactorEnabled);

```

Reply:
That looks great!